routes/POST-reset.js

```diff
@@ -0,0 +1,66 @@
+require("path");
+const fs = require("fs");
+const { promisify } = require("util");
+const handlebars = require("handlebars");
+const readFile = promisify(fs.readFile);
+const crypto = require("crypto")
+
+// Reset Password Request
+module.exports = (app, db, _, transporter) => {
+	app.post("/reset", (req, res) => {
+		const email = req.body.email;
+		if(email === undefined) {
+			return res.status(400).json({ error: "Missing fields, check our API docs at cloudclub.ca/api"})
+		}
+		db.query(`SELECT * FROM cloudclub.logins WHERE email='${email}'`, (err1, result) => {
+			if (err1) { console.log(err1); return res.status(500).json({ error: "Internal Server Error 500" }); }
+			if (result.length != 0) {
+				// Get password reset requests from user
+				db.query(`SELECT * FROM cloudclub.password_reset WHERE user_id='${result[0].id}' ORDER BY request_timestamp DESC`, (err2, result2) => {
+					if (err2) { console.log(err2); return res.status(500).json({ error: "Internal Server Error 500" }); }
+
+					if (result2.length != 0) {
+						// Time between present and most recent request: 
+						var diff = Math.abs(new Date() - new Date(result2[0].request_timestamp));
+						var minutes = Math.floor((diff/1000)/60);
+						if (minutes < 15) { return res.status(201).json({ success: "An email has been sent if the email address is associated with an account." }); }
+					}
+					// Create a new reset request
+					var token;
+					var tokenLink;
+					crypto.randomBytes(128, function(ex, buf) {
+						token = buf.toString("hex");
```

Comment:
Do we know the maximum length of the character string generated in this case? If yes, then what is it?

This information is important to know because of the requirements to store this text in the database. It should also be noted that large strings are being compared in the database and that there is an overhead associated when querying for tokens. 

Please note that previously I mentioned migrating the database to suit larger character strings; however, this may not be necessary if the length of the randomBytes to be generated is reduced to 32.